internal/groups/api/authrequests.go

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package api
+
+import (
+	"context"
+
+	"github.com/absmach/magistrala"
+	"github.com/absmach/magistrala/internal/api"
+	"github.com/absmach/magistrala/pkg/apiutil"
+	"github.com/absmach/magistrala/pkg/auth"
+	"github.com/absmach/magistrala/pkg/errors"
+	svcerr "github.com/absmach/magistrala/pkg/errors/service"
+	"github.com/absmach/magistrala/pkg/policies"
+)
+
+func CreateGroupAuthReq(ctx context.Context, request interface{}) ([]*magistrala.AuthorizeReq, error) {
+	req := request.(createGroupReq)
+	if err := req.validate(); err != nil {
+		return nil, errors.Wrap(apiutil.ErrValidation, err)
+	}
+
+	session, ok := ctx.Value(api.SessionKey).(auth.Session)
+	if !ok {
+		return nil, svcerr.ErrAuthorization
+	}
+
+	prs := []*magistrala.AuthorizeReq{
+		{
+			SubjectType: policies.UserType,
+			SubjectKind: policies.UsersKind,
+			Subject:     session.DomainUserID,
+			Permission:  policies.CreatePermission,
+			ObjectType:  policies.DomainType,
+			Object:      session.DomainID,
+		},
+	}
+
+	if req.Group.Parent != "" {
+		prs = append(prs, &magistrala.AuthorizeReq{
+			SubjectType: policies.UserType,
+			SubjectKind: policies.TokenKind,
+			Subject:     req.token,
+			Permission:  policies.EditPermission,
+			ObjectType:  policies.GroupType,
+			Object:      req.Group.Parent,
+		})
+	}
+
+	return prs, nil
+}
+
+func ViewGroupAuthReq(_ context.Context, request interface{}) ([]*magistrala.AuthorizeReq, error) {
```

Comment:
Why are we using slices with only a single element in these methods?

Reply:
There are some methods that have more than one authorize request